### DISCUSSION TITLE
If do_not_merge_across_partitions_select_final is used, can i still have merges into partition

### DISCUSSION THREAD

**Discussion by raghukn** [ClickHouse/ClickHouse#68511] [General]
I am having a system which has mutating data (Leads) atleast created_on field do not change over time for the records. I have implemented a ReplacingMergeTree MV to ensure a single row for a given Lead is ensured after deduplication. Now since I will have penalty of FINAL in my queries I am trying to alteast use do_not_merge_across_partitions_select_final to reduce the time of my FINAL query.

Question:
Does use of do_not_merge_across_partitions_select_final mean now mergerges happen to partitions that are once merged? Meanign if new Leads rows come via CDC of partition which was 5 months old , does the merge of that be prevented by do_not_merge_across_partitions_select_final?

If do_not_merge_across_partitions_select_final is ONLY Across partition merges, then I might be able to use this.

**Comment by den-crane** [comment]
--do_not_merge_across_partitions_select_final arg     Merge parts only in one partition in **select final**

https://fiddle.clickhouse.com/633e148b-c7f0-408c-b565-97a237785bd8

https://clickhouse.com/docs/en/guides/replacing-merge-tree#exploiting-partitions-with-replacingmergetree
